### DISCUSSION TITLE
Calling your own phone?

### DISCUSSION THREAD

**Discussion by xpseudonym** [juha-h/baresip-studio#488] [General]
When calling from your own mobile should I expect an 'engaged/ busy' response?
That is, if I attempt to test my SIP account by calling it from my mobile phone which hosts`baresip`, should `baresip` provide a busy/ engaged signal or ringing signal?
Thanks

**Comment by juha-h** [comment]
I don't quite understand your question, but if I call from my baresip account the same account, I get "Auto-rejected incoming call" and "486 Busy" response was sent.

**Comment by gdt** [comment]
I think the question was a call on the same device between baresip/SIP and a Dialer/simcard acccount. 

But it was an odd question, since @xpseudonym did not say what happened when they tried it, and how that didn't meet expecations.

**Reply by xpseudonym** [reply]
Thanks @gdt for deciphering my cryptics...
Yes, 'a call on the same device between baresip/SIP and a Dialer/simcard acccount' - but, more particularly, outgoing call on Dialer/simcard acccount to incoming call on baresip/SIP
ATM, when I try the above I'm getting 'busy tone' from my SIP provider - it's just a quick and easy way to check it my SIP provider is providing me a service...  I'm trying to debug where a failure is.

**Comment by juha-h** [comment]
Outgoing call fails if baresip cannot get exclusive audio focus.  I would assume that it is the case if a cellular call already exists.

**Reply by xpseudonym** [reply]
Thanks @juha-h that makes sense - and, does incoming call also fail if baresip cannot get audio focus - and likely to return 'busy'?

**Comment by juha-h** [comment]
I just tried to call my baresip account on my mobile device from baresip running on my Linux host when cellular call was active on the mobile device.  This is what I got:
```
call: connecting to 'sip:foo@test.tutpro.com'..
stream: audio: starting mediaenc 'zrtp' (wait_secure=0)
stream: video: starting mediaenc 'zrtp' (wait_secure=0)
call id: 1264b7c12e0768f7
call: SIP Progress: 100 Trying (/)
call: SIP Progress: 100 Trying (/)
call: SIP Progress: 180 Ringing (/)
sip:foo@test.tutpro.com: session closed: 486 Busy Here
```
adb logcat showed this:
```
07-18 19:05:11.453 17035 17098 D Baresip : Requesting audio focus
07-18 19:05:11.466  1639  4846 I MediaFocusControl: requestAudioFocus() from uid/pid 10516/17035 AA=USAGE_VOICE_COMMUNICATION/CONTENT_TYPE_SPEECH clientId=android.media.AudioManager@fc4466fcom.tutpro.baresip.BaresipService$Companion$$ExternalSyntheticLambda0@ce1ff7c callingPack=com.tutpro.baresip req=4 flags=0x0 sdk=34
07-18 19:05:11.469 17035 17098 W Baresip : requestAudioFocus denied
07-18 19:05:11.470 17035 17098 D Baresip : Auto-rejecting incoming call -5476376658622437424/-5476376658728028144/sip:test@test.tutpro.com
07-18 19:05:11.470 17035 17098 D Baresip Lib: hanging up call -5476376658622437424/-5476376658728028144
07-18 19:05:11.471 17035 17099 D Baresip Lib: call: rejecting incoming call from sip:test@test.tutpro.com (486 Busy Here)
```
So it was as I suspected: audio focus was denied and baresip app replied `486 Busy Here`.

There is bug https://github.com/baresip/baresip/issues/3061  in baresip lib and app is not able to prevent `180 Ringing` response before  `486 Busy Here` response.

Is there still something unanswered related to this discussion?

**Comment by xpseudonym** [comment]
Thanks @juha-h for taking the time to look closely at this - I'll stop trying to call my number from my mobile (which hosts the app) as a quick and dirt test... (and understand why it always responds with 'busy')
M
